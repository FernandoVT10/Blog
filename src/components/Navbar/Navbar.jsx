import Link from "next/link";

import "./Navbar.scss";

export default () => {
    return (
        <header>
            <nav className="navbar navbar-expand-md navbar-dark">
                <Link href="/">
                    <a className="navbar-brand navbar__brand">
                        <img
                        src="/favicon.ico"
                        width="50"
                        height="50"
                        className="d-inline-block align-top rounded-circle"
                        alt="Icon" />
                        <h1 className="navbar__title">
                            Fernando Vaca Tamayo
                        </h1>
                    </a>
                </Link>

                <button
                className="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#main-navbar"
                aria-controls="main-navbar"
                aria-expanded="false"
                aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse navbar__items" id="main-navbar">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <Link href="/">
                                <a className="nav-link navbar__link">Home</a>
                            </Link>
                        </li>
                        <li className="nav-item">
                            <Link href="/articles">
                                <a className="nav-link navbar__link">Articles</a>
                            </Link>
                        </li>
                        <li className="nav-item">
                            <Link href="/projects">
                                <a className="nav-link navbar__link">Projects</a>
                            </Link>
                        </li>
                        <li className="nav-item">
                            <Link href="/contact-me">
                                <a className="nav-link navbar__link">Contact me</a>
                            </Link>
                        </li>
                    </ul>
                </div>
            </nav>
        </header>
    );
};