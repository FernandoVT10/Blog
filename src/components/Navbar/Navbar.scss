@use "../../styles/variables";

.navbar {
    padding: 0 30px;
    background: variables.$primary;

    &__brand {
        display: flex;
        align-items: center;
    }

    &__title {
        font-family: 'Pacifico', cursive;
        font-size: var(--subtitle);
        margin: 0;
        margin-left: 10px;
    }

    &__items {
        justify-content: flex-end;
    }

    &__link {
        font-size: 14px;
    }
}