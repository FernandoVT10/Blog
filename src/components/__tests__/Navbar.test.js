import Navbar from "../Navbar/Navbar";
import { create } from "react-test-renderer";

test("Check if <Navbar/> renders correctly", () => {
    const component = create(<Navbar/>);

    expect(component.toJSON()).toMatchSnapshot();
});