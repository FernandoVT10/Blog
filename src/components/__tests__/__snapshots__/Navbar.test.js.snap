// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <Navbar/> renders correctly 1`] = `
<header>
  <nav
    className="navbar navbar-expand-md navbar-dark"
  >
    <a
      className="navbar-brand navbar__brand"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <img
        alt="Icon"
        className="d-inline-block align-top rounded-circle"
        height="50"
        src="/favicon.ico"
        width="50"
      />
      <h1
        className="navbar__title"
      >
        Fernando Vaca Tamayo
      </h1>
    </a>
    <button
      aria-controls="main-navbar"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#main-navbar"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse navbar__items"
      id="main-navbar"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link navbar__link"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link navbar__link"
            href="/articles"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Articles
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link navbar__link"
            href="/projects"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Projects
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link navbar__link"
            href="/contact-me"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Contact me
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
